services/QuillLMS/client/app/bundles/Evidence/helpers/containerRenderHelpers.tsx

```diff
@@ -131,6 +143,7 @@ export const renderPromptSteps = ({
   return (<div className="prompt-steps">
     {renderDirectionsSectionAndModal({ className: 'hide-on-mobile', closeReadTheDirectionsModal, activeStep, doneHighlighting, showReadTheDirectionsModal, activities })}
     {steps}
+    {activityIsComplete && renderCompletionButton()}
```

Comment:
I think you should pass the `completionButtonCallback` and the `activityIsComplete` props into `PromptStep` and then use them [here](https://github.com/empirical-org/Empirical-Core/blame/production/services/QuillLMS/client/app/bundles/Evidence/components/studentView/promptStep.tsx#L293). It looks like the logic at one point was supposed to do something similar with the `everyOtherStepCompleted ? 'Done : ...` thing but evidently that stopped working at some point. It might be worth just trying to fix that logic, but alternately in there you could just change the `onClick` handler based on the value of `activityIsComplete` and then you wouldn't have to render this whole other button.

Reply:
Good idea - addressed.